core/src/main/java/it/gov/pagopa/rtd/payment_instrument_manager/service/PaymentInstrumentManagerServiceImpl.java

```diff
@@ -116,28 +116,35 @@ public void generateFileForAcquirer() {
 
     public void refreshActiveHpans() {
 
-        if (log.isInfoEnabled()) {
-            log.info("PaymentInstrumentManagerServiceImpl.refreshActiveHpans");
-        }
-
-        Map<String,Object> awardPeriodData = paymentInstrumentManagerDao.getAwardPeriods();
-        String startDate = String.valueOf(awardPeriodData.get("start_date"));
-        String endDate = String.valueOf(awardPeriodData.get("end_date"));
-
-        OffsetDateTime saveExecutionDate = OffsetDateTime.now();
-        String saveExecutionDateString = saveExecutionDate.toString();
-
-        Map<String,Object> executionDates = paymentInstrumentManagerDao.getRtdExecutionDate();
-
-        writeBpdHpansToRtd(String.valueOf(executionDates.get("bpd_exec_date")), startDate, endDate);
+      OffsetDateTime saveExecutionDate = OffsetDateTime.now();
+      String saveExecutionDateString = saveExecutionDate.toString();
+
+      if (log.isInfoEnabled()) {
+          log.info("PaymentInstrumentManagerServiceImpl.refreshActiveHpans");
+      }
+
+      Map<String, Object> awardPeriodData = paymentInstrumentManagerDao.getAwardPeriods();
+      Object tmpStartDate = awardPeriodData.get("start_date");
+      Object tmpEndDate = awardPeriodData.get("end_date");
+
+      // During phaseout the award period can be null.
+      // In this case start and end date are null, hence no update is done
+      if (tmpStartDate!=null && tmpEndDate!=null){
+        String startDate = String.valueOf(tmpStartDate);
+        String endDate = String.valueOf(tmpEndDate);
+
+        Map<String, Object> executionDates = paymentInstrumentManagerDao.getRtdExecutionDate();
+        writeBpdHpansToRtd(String.valueOf(executionDates.get("bpd_exec_date")), startDate,
+            endDate);
         writeFaHpansToRtd(String.valueOf(executionDates.get("fa_exec_date")));
-        disableBpdHpans(String.valueOf(executionDates.get("bpd_del_exec_date")),startDate);
+        disableBpdHpans(String.valueOf(executionDates.get("bpd_del_exec_date")), startDate);
```

Comment:
I'm not sure this will work as intended. If I understand it correctly you are preventing _any_ update in case of missing award period, not only the update of BPD related data. I think that even when no award period is open is necessary to update FA HPANS and eventually invoke the final deleteDisabledHpans() method. What do you think?

Reply:
Right, thanks for the observation.
Fixed in the new commit.